lib/client/api.go

```diff
@@ -2520,12 +2539,20 @@ func (tc *TeleportClient) GetAppSession(ctx context.Context, req types.GetAppSes
 	)
 	defer span.End()
 
-	proxyClient, err := tc.ConnectToProxy(ctx)
+	clusterClient, err := tc.ConnectToCluster(ctx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	defer proxyClient.Close()
-	return proxyClient.GetAppSession(ctx, req)
+	defer clusterClient.Close()
+
+	rootAuthClient, err := clusterClient.ConnectToRootCluster(ctx)
```

Comment:
It looks like some of these change in semantic, which I assume is on purpose. Should we mention that in the PR description?